lib/logsource.c

```diff
@@ -633,7 +633,20 @@ log_source_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_options
                   evt_tag_printf("msg", "%p", msg));
 
   msg_set_context(NULL);
+}
+
+static void
+_initialize_dynamic_window(LogSource *self, gint init_window_size)
+{
+  self->dynamic_window_initialized = TRUE;
+  window_size_counter_set(&self->window_size, init_window_size);
+  self->full_window_size = init_window_size;
```

Comment:
`self->window_size` and `self->full_window_size` are used even when dynamic window is disabled. I'd rename these methods and the field to `_initialize_window`, `_is_window_initialized` and `LogSource::window_initialized`.

Reply:
Indeed.